### PR TITLE
chore(style) migrate to ESLint. closes #12

### LINKED ISSUE #12: Conform to brightspace eslint

### PR DESCRIPTION
**Pull request by j3parker**
Had to disable the no-invalid-this rule in two files. Removing the "this" was also problematic. Totally don't know what the right thing is here :)

### CODE REVIEW

**Comment by j3parker** [comment]
Maybe we should disable this rule in the Brightspace config... working on something else and it got triggered there too but I understand that things context :) What do you think?

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "brightspace",
+  "env": {
```
Comment (omsmith) [inline .eslintrc]: I think there's `brightspace/node-confug`

**Comment by omsmith** [comment]
Just need to put a `/* @this */` before those two functions, to indicate it's legit there

File: README.md [hunk 2]
```diff
@@ -62,11 +62,10 @@ private parameters are not available.
...
-and [JSHint][JSHint] rules.
+This repository is configured with [EditorConfig][EditorConfig] and
+[ESLint][ESLint] rules.
 
 [algs]: https://tools.ietf.org/html/rfc7518#section-3.1
 [jwk]: https://tools.ietf.org/html/rfc7517
 [EditorConfig]: http://editorconfig.org/
-[jscs]: http://jscs.info/
-[JSHint]: http://jshint.com/
+[ESLint]: https://github.com/eslint/eslint
```
Comment (omsmith) [inline README.md]: http://eslint.org/

File: package.json [hunk 3]
```diff
@@ -31,6 +33,8 @@
   "devDependencies": {
     "chai": "^3.4.0",
     "coveralls": "^2.11.4",
+    "eslint": "^1.8.0",
```
Comment (omsmith) [inline package.json]: `npm install eslint@latest --save-dev`

File: package.json [hunk 4]
```diff
@@ -31,6 +33,8 @@
...
+    "eslint-config-brightspace": "0.0.3",
```
Comment (omsmith) [inline package.json]: `npm install eslint-config-brightspace@latest --save-dev`

Comment (j3parker) [inline package.json]: done, thanks. I was manually adding these because I wasn't using npm right :)

File: package.json (hunk 3 above)
Comment (j3parker) [inline package.json]: done

File: README.md (hunk 2 above)
Comment (j3parker) [inline README.md]: done

**Comment by j3parker** [comment]
thanks for the `/* @this */` tip - done.

File: .eslintrc (hunk 1 above)
Comment (j3parker) [inline .eslintrc]: I don't follow

Comment (omsmith) [inline .eslintrc]: https://github.com/Brightspace/nodejs-microservice-skeleton/blob/master/.eslintrc

Comment (j3parker) [inline .eslintrc]: Sweet, done

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1 @@
+.gitignore
```
Comment (omsmith) [inline .eslintignore]: this is a symlink right?

Comment (j3parker) [inline .eslintignore]: yeah, it's sorta possible to tell in github but its not obvious

Comment (j3parker) [inline .eslintignore]: if you click "view" on the file it says so

**Comment by omsmith** [comment]
LGTM

**Pull request merged**